src/Workspaces/Core/Portable/Shared/Extensions/ITypeParameterSymbolExtensions.cs

```diff
@@ -20,5 +20,15 @@ private static INamedTypeSymbol GetNamedTypeSymbol(ITypeSymbol type)
                     ? GetNamedTypeSymbolConstraint((ITypeParameterSymbol)type)
                     : null;
         }
+
+        public static bool IsAccessibleFromSymbolOrEnclosingType(this ITypeParameterSymbol typeParameter, ISymbol within)
+        {
+            if (typeParameter.DeclaringMethod != null)
+            {
+                return typeParameter.DeclaringMethod == within;
```

Comment:
you might be in a nested local function inside that method (add test for this plz).